state/model.go

```diff
@@ -454,6 +454,14 @@ func (ctlr *Controller) NewModel(args ModelArgs) (_ *Model, _ *State, err error)
 		return nil, nil, errors.Annotate(err, "could not start state for new model")
 	}
 
+	// Validate the constraints of a model cam only be done after the state
```

Comment:
We can actually do a little refactoring to avoid having to have a state instance to do the constraints validation. We know the cloud, we can create a the relevant state policy etc without needing to create the state.